tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py

```diff
@@ -0,0 +1,863 @@
+    # results = SyscollectorScansTestsResults()
+    results = {}
+
+    @pytest.fixture(scope='class')
+    def get_results(self):
+        return self.results
+
+    def test_syscollector_first_scan(self, request, host_manager, setup_vulnerability_tests, get_results):
+        """
+        description: Validates the initiation of Syscollector scans across all agents in the environment.
+
+        This test ensures that the Vulnerability Detector accurately detects vulnerabilities within the environment.
+        The Agent's Vulnerability Indexer index is expected to be updated with the detected vulnerabilities.
+
+        tier: 0
+
+        parameters:
+            - request: pytest request object
+            - host_manager:
+                type: fixture
+                brief: Get the host manager of the environment
+            - setup_vulnerability_tests:
+                type: fixture
+                brief: Setup the environment to proceed with the testing
+            - get_results: fixture to get the results of global class tests
+
+        assertions:
+            - Verify that syscollector scan is started after agent start in all agents
+
+        cases: None
+
+        tags:
+            - syscollector
+            - vulnerability_detector
+        """
+        results = get_results
+        test_name = request.node.name
+
+        test_result = {
+                'checks': {
+                    'all_successfull': True,
+                },
+                'evidences': {
+                    'agents_not_scanned_first_scan': []
+                }
+        }
+
+        logger.critical("Monitoring Syscollector First Scan")
+        list_hosts = host_manager.get_group_hosts('agent')
+        monitoring_data = generate_monitoring_logs(host_manager,
+                                                   [get_event_regex({'event': 'syscollector_scan_start'}),
+                                                    get_event_regex({'event': 'syscollector_scan_end'})],
+                                                   [TIMEOUT_SYSCOLLECTOR_SCAN, TIMEOUT_SYSCOLLECTOR_SCAN],
+                                                   list_hosts)
+        monitoring_results = monitoring_events_multihost(host_manager, monitoring_data)
+
+        logger.critical(f"Value of monitoring results is: {monitoring_results}")
+
+        for agent in monitoring_results:
+            if monitoring_results[agent]['not_found']:
+                test_result['checks']['all_successfull'] = False
+                test_result['evidences']['agents_not_scanned_first_scan'].append(agent)
+
+        results[test_name] = test_result
+
+        if not test_result['checks']['all_successfull']:
+            pytest.fail("Some agents has not been scanned:"
+                        f"{test_result['evidences']['agents_not_scanned_first_scan']}."
+                        "Check logs for more information")
+        else:
+            logger.critical("All agents has been scanned")
```

Comment:
```suggestion
        assert test_result['checks']['all_successfull'], f"Some agents has not been scanned: {test_result['evidences']['agents_not_scanned_first_scan']}."
```

Reply:
Is important to store the evidences in order to see it easily through the report even if this make less readable the tests.
We should design general classes a libraries to handle tests results